### PR TITLE
WIP #1363 bi-directional transports

### PR DESCRIPTION
**Pull request by msevestre**

### CODE REVIEW

File: src/PKSim.Assets/PKSimConstants.cs [hunk 1]
```diff
@@ -2621,26 +2625,5 @@ public static class Comparison
          public static readonly string Absent = "Absent";
          public static readonly string Present = "Present";
       }
-
-      public static class TransportDirection
```
Comment (msevestre) [inline src/PKSim.Assets/PKSimConstants.cs:2625]: This was not required anymore (we have an enum)

File: src/PKSim.Assets/PKSimConstants.cs [hunk 2]
```diff
@@ -900,6 +900,10 @@ public static string MapToSnapshotNotSupportedWithoutContext(string snapshotType
          public static string AlteredBuildingBlockNotFoundInSimulation(string simulationName, string buildingBlockName, string buildingBlockType) =>
             $"Could not update the altered flag for {buildingBlockType} building block '{buildingBlockName}' as it is not used in {ObjectTypes.Simulation} '{simulationName}'.";
 
+
+         public static string CannotCreateTransportProcessWithKinetic(string processName, string kineticType) =>
```
Comment (msevestre) [inline src/PKSim.Assets/PKSimConstants.cs:904]: This is the message shown when selecting BiDirectional and HILL

File: src/PKSim.Core/Model/TransportDirection.cs [hunk 3]
```diff
@@ -10,27 +10,33 @@ public enum TransportDirectionId
       None,
       InfluxPlasmaToInterstitial,
       EffluxInterstitialToPlasma,
+      BiDirectionalPlasmaInterstitial,
```
Comment (msevestre) [inline src/PKSim.Core/Model/TransportDirection.cs:13]: This is really ther only change to implement in PKSim. ALl the rest is only DB

**Review by msevestre — COMMENTED** [review]
There is a specific issue for Display name and description #1681

**Comment by msevestre** [comment]
I need to create a new TransportType in core (BiDirectional) so that we can also add it to the default selection to set all values at once. Aside from that, it's all there and working as far as I can tell

File: src/PKSim.Core/Repositories/ISimulationActiveProcessRepository.cs [hunk 4]
```diff
@@ -22,7 +22,7 @@ public interface ISimulationActiveProcessRepository : IStartableRepository<IPKSi
       TActiveProcess ProcessFor<TActiveProcess>(string processName) where TActiveProcess : IPKSimProcess;
 
       /// <summary>
-      ///    Returns the transport compoundProcess for the given compound compoundProcess
+      ///    Returns the transport transport process for the given <paramref name="individualProcessName"/> and <paramref name="compoundProcessName"/>
```
Comment (PavelBal) [inline src/PKSim.Core/Repositories/ISimulationActiveProcessRepository.cs:25]: One "transport" too many?

**Review by PavelBal — COMMENTED** [review]
Look fine for me, but I guess @Yuri05 should also review before merging.

File: src/Db/Dump/PKSimDB_dump.txt [hunk 5]
```diff
@@ -4497,6 +4505,7 @@ INSERT INTO tab_calculation_method_rates VALUES('LinksCommon','ActiveEffluxSpeci
 INSERT INTO tab_calculation_method_rates VALUES('LinksCommon','ActiveEffluxSpecificWithTransporterInSource_MM','TM * kcat * K_water_src * C_src / (Km + K_water_src * C_src)','Amount per time');
 INSERT INTO tab_calculation_method_rates VALUES('LinksCommon','ActiveInfluxSpecificWithTransporterInTarget_Hill','TM * kcat * (K_water_src * max(C_src;0))^alpha / (Km^alpha + (K_water_src * max(C_src;0))^alpha)','Amount per time');
 INSERT INTO tab_calculation_method_rates VALUES('LinksCommon','ActiveInfluxSpecificWithTransporterInTarget_MM','TM * kcat * K_water_src * C_src / (Km + K_water_src * C_src)','Amount per time');
+INSERT INTO tab_calculation_method_rates VALUES('LinksCommon','ActiveBiDirectionalSpecific_MM','TM * kcat * (C_src * K_water_src - C_target * K_water_target) / (Km + C_src * K_water_src + C_target * K_water_target)','Amount per time');
```
Comment (Yuri05) [inline src/Db/Dump/PKSimDB_dump.txt:4508]: @PavelBal I don't know if the formula is correct. If you say "yes" - I'll just trust you :)

**Review by Yuri05 — APPROVED** [review]
if the formula for the directional transport is correct - can be merged
(TODOs in descriptions can be fixed later)

File: src/PKSim.Core/Repositories/ISimulationActiveProcessRepository.cs (hunk 4 above)
Comment (msevestre) [reply src/PKSim.Core/Repositories/ISimulationActiveProcessRepository.cs:25]: good catch. Will fix in next PR

**Pull request merged**